diff --git a/src/Composer/Installers/Installer.php b/src/Composer/Installers/Installer.php
--- a/src/Composer/Installers/Installer.php
+++ b/src/Composer/Installers/Installer.php
@@ -136,8 +136,8 @@
         IOInterface $io,
         Composer $composer,
         $type = 'library',
-        Filesystem $filesystem = null,
-        BinaryInstaller $binaryInstaller = null
+        ?Filesystem $filesystem = null,
+        ?BinaryInstaller $binaryInstaller = null
     ) {
         parent::__construct($io, $composer, $type, $filesystem,
             $binaryInstaller);
diff --git a/src/Composer/Installers/BaseInstaller.php b/src/Composer/Installers/BaseInstaller.php
--- a/src/Composer/Installers/BaseInstaller.php
+++ b/src/Composer/Installers/BaseInstaller.php
@@ -20,7 +20,7 @@
      * @param Composer         $composer
      * @param IOInterface      $io
      */
-    public function __construct(PackageInterface $package = null, Composer $composer = null, IOInterface $io = null)
+    public function __construct(?PackageInterface $package = null, ?Composer $composer = null, ?IOInterface $io = null)
     {
         $this->composer = $composer;
         $this->package = $package;
