diff --git a/media_entity_soundcloud.module b/media_entity_soundcloud.module
index 22b9789..0c40587 100644
--- a/media_entity_soundcloud.module
+++ b/media_entity_soundcloud.module
@@ -13,6 +13,7 @@ function media_entity_soundcloud_theme($existing, $type, $theme, $path) {
     'media_soundcloud_embed' => [
       'variables' => [
         'source_id' => NULL,
+        'secret_token' => NULL,
         'width' => NULL,
         'height' => NULL,
         'type' => NULL,
diff --git a/media_entity_soundcloud.theme.inc b/media_entity_soundcloud.theme.inc
index 968ced5..f4b9975 100644
--- a/media_entity_soundcloud.theme.inc
+++ b/media_entity_soundcloud.theme.inc
@@ -32,8 +32,13 @@ function template_preprocess_media_soundcloud_embed(&$variables) {
   // Add color.
   if (!empty($variables['color'])) {
     $query['color'] = $variables['color'];
   }

+  // Add secret_token for private URLs
+  if(!empty($variables['secret_token'])) {
+    $query['secret_token'] = $variables['secret_token'];
+  }
+
   $url = Url::fromUri($uri, ['query' => $query]);
   $variables['url'] = $url->toString();
 }
diff --git a/src/Plugin/Field/FieldFormatter/SoundcloudEmbedFormatter.php b/src/Plugin/Field/FieldFormatter/SoundcloudEmbedFormatter.php
index 66e9f4f..3dc7099 100644
--- a/src/Plugin/Field/FieldFormatter/SoundcloudEmbedFormatter.php
+++ b/src/Plugin/Field/FieldFormatter/SoundcloudEmbedFormatter.php
@@ -116,6 +116,7 @@ class SoundcloudEmbedFormatter extends FormatterBase {
           $element[$delta] = [
             '#theme' => 'media_soundcloud_embed',
             '#source_id' => $source_id,
+            '#secret_token' =>  $source->getMetadata($media, 'secret_token'),
             '#width' => $this->getSetting('width'),
             '#height' => $this->getSetting('height'),
             '#type' => $this->getSetting('type'),
diff --git a/src/Plugin/media/Source/Soundcloud.php b/src/Plugin/media/Source/Soundcloud.php
index 49dc235..0921649 100644
--- a/src/Plugin/media/Source/Soundcloud.php
+++ b/src/Plugin/media/Source/Soundcloud.php
@@ -147,6 +147,20 @@ class Soundcloud extends MediaSourceBase {
         }
 
         return FALSE;
+      case 'secret_token':
+        // Extract the src attribute from the html code.
+        preg_match('/src="([^"]+)"/', $data['html'], $src_matches);
+        if (!count($src_matches)) {
+          return FALSE;
+        }
+
+        // Extract the secret_token from the src.
+        preg_match('#&(secret_token)=([^&]+)#', urldecode($src_matches[1]), $matches);
+        if (!count($matches)) {
+          return FALSE;
+        }
+
+        return $matches[2];
 
       default:
         return parent::getMetadata($media, $attribute_name);
