diff --git a/message_digest.module b/message_digest.module
--- a/message_digest.module
+++ b/message_digest.module
@@ -134,7 +134,7 @@
 /**
  * Implements callback_allowed_values_function().
  */
-function message_digest_allowed_values_callback(FieldStorageDefinitionInterface $definition, FieldableEntityInterface $entity = NULL, &$cacheable = NULL) {
+function message_digest_allowed_values_callback(FieldStorageDefinitionInterface $definition, ?FieldableEntityInterface $entity = NULL, &$cacheable = NULL) {
   // Always add an 'immediate' option.
   $values = [
     t('Send immediately'),
