diff --git a/message_digest_ui/src/Plugin/Action/DigestInterval.php b/message_digest_ui/src/Plugin/Action/DigestInterval.php
--- a/message_digest_ui/src/Plugin/Action/DigestInterval.php
+++ b/message_digest_ui/src/Plugin/Action/DigestInterval.php
@@ -87,7 +87,7 @@
   /**
    * {@inheritdoc}
    */
-  public function access($object, AccountInterface $account = NULL, $return_as_object = FALSE) {
+  public function access($object, ?AccountInterface $account = NULL, $return_as_object = FALSE) {
     if (!$this->flag) {
       return $return_as_object ? AccessResult::forbidden() : FALSE;
     }
